format/FlightSql.proto

```diff
@@ -1797,6 +1797,27 @@ message DoPutUpdateResult {
   int64 record_count = 1;
 }
 
+/* An *optional* response returned when `DoPut` is called with `CommandPreparedStatementQuery`.
+ *
+ * *Note on legacy behavior*: previous versions of the protocol did not return any result for
+ * this command, and that behavior should still be supported by clients. See documentation
+ * of individual fields for more details on expected client behavior in this case. 
```

Comment:
> Maybe we could phrase this in terms of what happens in the case where clients or servers don't support this new message

I wanted to avoid documenting information about individual fields in the message, since the message might one day contain many fields.

I think I could simplify the wording on this to just indicate that older servers dont use this, clients should support that, and just use the default blank values for fields in that case.